Modules/Core/Common/include/itkImageAlgorithm.hxx

```diff
@@ -36,8 +36,8 @@ ImageAlgorithm::DispatchedCopy(const InputImageType *                       inIm
 {
   if (inRegion.GetSize()[0] == outRegion.GetSize()[0])
   {
-    itk::ImageScanlineConstIterator<InputImageType> it(inImage, inRegion);
-    itk::ImageScanlineIterator<OutputImageType>     ot(outImage, outRegion);
+    itk::ImageScanlineConstIterator it(inImage, inRegion);
```

Comment:
I assume this can be done for other iterator types? It would save a lot of typing. And, of course, if someone wants to explicitly provide the image type, they can do so.

Reply:
Note that for the old ITK iterator types, a deduction guide is required, in order to support deduction from a `SmartPointer<ImageType>`. Whereas the deduction guides of ImageBufferRange and ImageRegionRange (added with PR #3998) are only there to suppress those warnings (`-Wctad-maybe-unsupported`).